todo_app/app.py

```diff
@@ -1,11 +1,38 @@
-from flask import Flask
-
+from flask import Flask, request,render_template
 from todo_app.flask_config import Config
+from todo_app.data.session_items import get_items, add_item, get_item, save_item, delete_item
+
+
 
 app = Flask(__name__)
 app.config.from_object(Config())
+  
 
-
-@app.route('/')
+@app.route('/', methods=['GET','POST'] )
 def index():
-    return 'Hello World!'
+        items = get_items()
+        return render_template ('index.html', my_items = items)
+        
+
+@app.route('/add_tasks', methods=['POST'])
+def add_tasks():
+        add_item(dict(request.form.items())['title'],dict(request.form.items())['priority'], dict(request.form.items())['status'])
+        items = get_items()
+        return render_template ('index.html', my_items = items)
+        
+    
+@app.route('/update_task_status', methods=['POST'])
+def update_task():
+        item = get_item(dict(request.form.items())['id'])
+        item.update({'status': dict(request.form.items())['status']})
+        save_item(item)
+        items = get_items()
+        return render_template ('index.html', my_items = items)
```

Comment:
Here you can see we have some duplication of the index page function, which means when we make a change to that function we need to remember to change all the other routes too. But in addition, we get some slightly weird behaviour, for example:
* If I create a to-do then try to refresh the page straight after, it tries to create another one
* If I create a to-do so that my URL bar shows `http://localhost:5000/add_tasks` and then click into the URL bar and hit enter to load that URL again, I get `Method Not Allowed` as a response, since I'm GET-ting a POST endpoint

A nicer solution would be redirecting the user back to the index page once the request has been handled, so we can keep the experience more consistent:
```python
from flask import <...>, redirect

...

@app.route('/update_task_status', methods=['POST'])
def update_task():
        item = get_item(dict(request.form.items())['id'])
        item.update({'status': dict(request.form.items())['status']})
        save_item(item)
        return redirect('/')

```

Reply:
I actually noticed this as I was refreshing the page, but I did not get to investigate a solution. I will try to implement this :)